src/tests/users/signup.test.js

```diff
@@ -1,30 +1,268 @@
 const expect = require('expect')
 const { request } = require('../../utils/test')
-const { UserFactory } = require('../../utils/factories/')
+const { UserFactory, QualificationFactory } = require('../../utils/factories/')
+const Qualification = require('../../models/qualifications')
+const User = require('../../models/user')
 
 describe('signup', () => {
   const fakeUser = UserFactory.generate()
+  const fakeUser2 = UserFactory.generate()
+  let users = []
+  let skills = []
+  let knowledge = []
+  let user = null
+  before(async () => {
+    for (let y = 0; y < 5; y++) {
+      const skill = await Qualification.create(
+        QualificationFactory.generateSkill()
+      )
+      skills.push(`"${skill._id}"`)
+      const knowledge1 = await Qualification.create(
+        QualificationFactory.generateKnowledge()
+      )
+      knowledge.push(`"${knowledge1._id}"`)
+    }
+    for (let x = 0; x <= 1; x++) {
+      let user = await UserFactory.generate()
+      user = await User.signup(user)
+      users.push(user)
+    }
+  })
+
+  it("should not signup if mentor is not certified or mentor's email is not verified", () => {
+    return request({
+      query: `
+        mutation {
+          signup(record: {
+              firstName: "${fakeUser.firstName}"
+              lastName: "${fakeUser.lastName}"
+              email: "${fakeUser.email}"
+              password: "${fakeUser.password}"  
+              username: "${fakeUser.username}"
+              mentor: "${users[0]._id}"
+              skills: [${skills}]
+              knowledge: [${knowledge}]
+          }) {
+            record {
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('errors')
+        expect(Array.isArray(res.body.errors)).toBe(true)
+      })
+      .expect(200)
+  })
+
+  it('should make user a mentor', () => {
+    return request({
+      query: `
+        mutation {
+          updateUserById (_id: "${users[0]._id}", record: {
+            mentorshipCertified: true
+          }) {
+            record {
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.updateUserById.record')
+        expect(res.body.data.updateUserById.record.email).toStrictEqual(
+          users[0].email
+        )
+      })
+      .expect(200)
+  })
+
+  it('should not verify email with incorrect verification code', () => {
+    return request({
+      query: `
+        mutation {
+          verifyEmail (verificationCode: "XYZ") {
+            record 
+            error {
+              message
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.verifyEmail.error.message')
+        expect(res.body.data.verifyEmail.error.message).toStrictEqual(
```

Comment:
This test is too tightly-coupled. It should just check if it throws an error regardless of the message - which might be changed later on.

Reply:
yes please change statement here, it is precisely testing for error type. shall I add error code instead? so that message can be changed later for that code.